cilium-dbg/cmd/bpf_multicast_subscribers.go

```diff
@@ -0,0 +1,308 @@
+				return
+			}
+			Fatalf("Failed to open multicast bpf maps: %s", err)
+		}
+
+		subscriberMap, err := groupV4Map.Lookup(groupAddr)
+		if err != nil {
+			Fatalf("Failed to lookup multicast group %s: %s", groupAddr, err)
+		}
+
+		subscriber := &maps_multicast.SubscriberV4{
+			SAddr:    subIP,
+			Ifindex:  uint32(link.Attrs().Index),
+			IsRemote: true,
+		}
+
+		if err := subscriberMap.Insert(subscriber); err != nil {
+			Fatalf("Failed to add subscriber %s to multicast group %s: %s", subIP, groupAddr, err)
+		}
+	},
+}
+
+var MulticastGroupSubscriberDelCmd = &cobra.Command{
+	Use:     "delete <group> <subscriber-address>",
+	Aliases: []string{"del"},
+	Short:   "Delete a subscriber from the given multicast group.",
+	Long: `Delete a subscriber from the given multicast group.
+To delete remote subscriber, following information is required:
+- group: multicast group address from which subscriber is deleted.
+- subscriber-addr: subscriber IP address
+For Example: 
+To delete a remote node 10.100.0.1 from multicast group 229.0.0.1, use the following command:
+	cilium-dbg bpf multicast subscriber delete 229.0.0.1 10.100.0.1
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium bpf multicast subscriber delete")
+
+		groupAddr, subIP, err := parseMulticastGroupSubscriberArgs(args)
+		if err != nil {
+			Fatalf("invalid argument: %s\n", err)
+		}
+
+		groupV4Map, err := maps_multicast.OpenGroupV4OuterMap(maps_multicast.GroupOuter4MapName)
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				fmt.Fprintln(os.Stderr, "Cannot find multicast bpf maps")
+				return
+			}
+			Fatalf("Failed to open multicast bpf maps: %s", err)
+		}
+
+		subscriberMap, err := groupV4Map.Lookup(groupAddr)
+		if err != nil {
+			Fatalf("Failed to lookup multicast group %s: %s", groupAddr, err)
+		}
+
+		if err := subscriberMap.Delete(subIP); err != nil {
+			Fatalf("Failed to delete subscriber %s from multicast group %s: %s", subIP, groupAddr, err)
+		}
+	},
+}
+
+// subscriberData is used to store the subscribers of a multicast group
+type subscriberData struct {
+	GroupAddr   netip.Addr
+	Subscribers []maps_multicast.SubscriberV4
+}
+
+func outputGroups(groupV4Map maps_multicast.GroupV4Map, groupAddrs []netip.Addr) {
+	var allGroups []subscriberData
+	var wg sync.WaitGroup
+	wg.Add(len(groupAddrs))
+	subDataCh := make(chan subscriberData, len(groupAddrs))
+
+	for _, groupAddr := range groupAddrs {
+		// concurrently get subscribers for each group
+		go func(groupAddr netip.Addr) {
+			defer wg.Done()
+			groupData := subscriberData{
+				GroupAddr: groupAddr,
+			}
+
+			subscriberMap, err := groupV4Map.Lookup(groupAddr)
+			if err != nil {
+				Fatalf("Failed to lookup multicast group %s: %s", groupAddr, err)
+			}
+
+			subscribers, err := subscriberMap.List()
+			if err != nil {
+				Fatalf("Failed to list multicast subscribers for group %s: %s", groupAddr, err)
+			}
+
+			for _, sub := range subscribers {
```

Comment:
Could we just use the subscribers list we get and set it on the groupData vs making our own copy?